crates/ty_python_semantic/src/types/class.rs

```diff
@@ -2229,7 +2237,9 @@ impl<'db> KnownClass {
         db: &'db dyn Db,
         specialization: impl IntoIterator<Item = Type<'db>>,
     ) -> Type<'db> {
-        let class_literal = self.to_class_literal(db).expect_class_literal();
+        let Type::ClassLiteral(class_literal) = self.to_class_literal(db) else {
+            return Type::unknown();
+        };
```

Comment:
this was necessary or some custom-typeshed tests started panicking because they couldn't find `Exception` and `ExceptionGroup` in the custom typeshed